.github/workflows/cd.yml

```diff
@@ -53,18 +53,28 @@ jobs:
     needs: [dist]
     runs-on: ubuntu-latest
     if: github.event_name == 'release' && github.event.action == 'published'
-    environment: pypi
+    environment:
+      name: pypi
+      url: https://pypi.org/p/particle
     permissions:
       id-token: write
-
+      attestations: write
 
     steps:
     - uses: actions/download-artifact@v4
       with:
         name: artifact
         path: dist
 
-    - uses: pypa/gh-action-pypi-publish@release/v1
+    - name: Generate artifact attestation for sdist and wheel
+      uses: actions/attest-build-provenance@1c608d11d69870c2092266b3f9a6f3abbf17002c # v1.4.3
```

Comment:
I hope all this commit hashes will not become a pain to update?

Reply:
Dependabot should do it. Though we can use any other form, but this is common for this action since it's security related.

Reply:
OK.